### DISCUSSION TITLE
Many To Many Realtion issue

### DISCUSSION THREAD

**Discussion by milon27** [prisma/prisma#8973] [Q&A] [answered]
```prisma
model post {
  pid      Int    @id @default(autoincrement())
  title    String
  post_tag tag[]  @relation("post_tag")
}

model tag {
  tid      Int    @id @default(autoincrement())
  name     String
  post_tag post[] @relation("post_tag")
}
```

for this schema prisma automatically created a table **_post_tag** , how i can define that table model in my schema?

**Comment by ryands17** [comment, marked as answer]
@milon27 👋 

You don't need to define that model in your schema as you can directly query for posts and tags. This table is handled automatically by Prisma.

Here is the explanation for the same: https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations

**Reply by milon27** [reply]
```

model post {
  pid      Int    @id @default(autoincrement())
  title    String
  tags tag[]  
}

model tag {
  tid      Int    @id @default(autoincrement())
  name     String
  posts post[] 
}


model tag_post{
  post       Post     @relation(fields: [post_id], references: [id])  
  post_id   Int // relation scalar field (used in the `@relation` attribute above)
  tag      tag   @relation(fields: [tag_id], references: [id])
  tag_id    Int // relation scalar field (used in the `@relation` attribute above)

  @@id([post_id, tag_id])
}
```

if we define this way:

while creating a post: 

```
prisma.post.create({
  data: {
    title: 'How to be Bob',
    tags: {
           connect: {
               //it require the post id also tag id, but if we don't define that relation model that time we can only pass the tag id and create a post new post will automatically assign in the realtion table. 
             //now my question : is my defined model contain something wrong?

           },
      ],
    },
  },
})

```

**Reply by ryands17** [reply]
Have a look at this: https://www.prisma.io/docs/support/help-articles/working-with-many-to-many-relations#explicit-relations

**Reply by emrysal** [reply]
I'm also struggling with this; I appreciate Prisma is trying to make it's own Many-Many relationship table but for the definition:

```javascript
_EventTypeToUser { A, B }
// I'd rather have 
eventtype_user { eventtype_id, user_id }
```

So OK, my next approach was the following:
```javascript
model EventTypeUser {
  eventTypeId Int       @map("eventtype_id")
  userId      Int       @map("user_id")
  eventType   EventType @relation(fields: [eventTypeId], references: [id])
  user        User      @relation(fields: [userId], references: [id])

  @@id([eventTypeId, userId])
  @@map("eventtype_user")
}
```

However, this changes all of my selects, instead of being able to do:

```javascript
1:
organizers: {
  select: {
    ...
  }
}

// now I have to:
2:
organizers: {
  select: {
    user: {
      select: {
        ...
      }
    }
  }
}
```

Is there a way to achieve the naming without converting to the much less nice (2) example?

**Reply by ryands17** [reply]
Unfortunately as you're using explicit many-to-many relations you need to go one level deeper to fetch the actual relation. There's a feature request for the same that you can follow and add a 👍 as well.
https://github.com/prisma/prisma/issues/5369

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
